prow/cmd/deck/main.go

```diff
@@ -1104,6 +1106,34 @@ func validateLogRequest(r *http.Request) error {
 	return nil
 }
 
+func handleProwJob(prowJobClient prowv1.ProwJobInterface) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		name := r.URL.Query().Get("prowjob")
+		if name == "" {
+			http.Error(w, "request did not provide the 'name' query parameter", http.StatusBadRequest)
+			return
+		}
+		pj, err := prowJobClient.Get(name, metav1.GetOptions{})
+		if err != nil {
+			http.Error(w, fmt.Sprintf("ProwJob not found: %v", err), http.StatusNotFound)
+			if !kerrors.IsNotFound(err) {
+				// admins only care about errors other than not found
+				logrus.WithError(err).Warning("ProwJob not found.")
+			}
+			return
+		}
+		b, err := yaml.Marshal(&pj)
```

Comment:
Technically we should do this with the scheme in case we end up with multiple versions of ProwJob in the future but likely overkill ...

Reply:
Could you clarify what you mean by scheme?

Reply:
It's overkill here but there is technically a more correct way to serialize internal ProwJobs into external ProwJobs. We should omit it for now